app/controllers/api/v1/search_controller.rb

```diff
@@ -0,0 +1,293 @@
+module Api
+  module V1
+    class SearchController < ApiBaseController
+      include Concerns::Authenticator
+      include Swagger::Blocks
+
+      before_action :set_current_api_user!
+      before_action :set_search_facet, only: :facet_filters
+      before_action :set_search_facets_and_filters, only: :index
+
+      swagger_path '/search' do
+        operation :get do
+          key :tags, [
+              'Search'
+          ]
+          key :summary, 'Faceted & keyword search for studies & cells'
+          key :description, 'Search studies or cells using facets and keywords'
+          key :operationId, 'search_studies'
+          parameter do
+            key :name, :type
+            key :in, :query
+            key :description, 'Type of query to perform (study- or cell-based)'
+            key :required, true
+            key :type, :string
+            key :enum, ['study', 'cell']
```

Comment:
I double-taked the lack of `'gene'`, but I suppose omitting it makes sense.

That said, I don't think we track yet the task of integrating the existing global gene search endpoint into the upcoming React front-end.

Reply:
Yes - global gene search functionality is totally separate from keyword/faceted search and should continue to be treated as such.  It requires a totally different query/response flow with different methods and views.